File: docs/07-archive/legacy-repos/ebs_bo/src/bo/schemas/player.py
from pydantic import BaseModel


class PlayerCreate(BaseModel):
    first_name: str
    last_name: str
    wsop_id: str | None = None
    nationality: str | None = None
    country_code: str | None = None
    profile_image: str | None = None


class PlayerUpdate(BaseModel):
    first_name: str | None = None
    last_name: str | None = None
    wsop_id: str | None = None
    nationality: str | None = None
    country_code: str | None = None
    profile_image: str | None = None
    player_status: str | None = None


class PlayerRead(BaseModel):
    player_id: int
    first_name: str
    last_name: str
    wsop_id: str | None
    nationality: str | None
    country_code: str | None
    player_status: str
    created_at: str
    updated_at: str

File: team1-frontend/lib/models/entities/player.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'player.freezed.dart';
part 'player.g.dart';

@freezed
class Player with _$Player {
  const factory Player({
    @JsonKey(name: 'playerId') required int playerId,
    @JsonKey(name: 'wsopId') String? wsopId,
    @JsonKey(name: 'firstName') required String firstName,
    @JsonKey(name: 'lastName') required String lastName,
    String? nationality,
    @JsonKey(name: 'countryCode') String? countryCode,
    @JsonKey(name: 'profileImage') String? profileImage,
    @JsonKey(name: 'playerStatus') required String playerStatus,
    @JsonKey(name: 'isDemo') @Default(false) bool isDemo,
    required String source,
    @JsonKey(name: 'syncedAt') String? syncedAt,
    @JsonKey(name: 'createdAt') required String createdAt,
    @JsonKey(name: 'updatedAt') required String updatedAt,
    int? stack,
    @JsonKey(name: 'tableName') String? tableName,
    @JsonKey(name: 'seatIndex') int? seatIndex,
  }) = _Player;

  factory Player.fromJson(Map<String, dynamic> json) =>
      _$PlayerFromJson(json);
}
